management-api-for-apache-cassandra.advisories.yaml

```diff
@@ -0,0 +1,102 @@
+schema-version: 2.0.2
+
+package:
+  name: management-api-for-apache-cassandra
+
+advisories:
+  - id: CVE-2020-36518
+    aliases:
+      - GHSA-57j2-w4cx-62h2
+    events:
+      - timestamp: 2024-01-14T18:15:03Z
+        type: pending-upstream-fix
+        data:
+          note: "To fix the CVE, we have to upgrade 'swagger-jaxrs2' to '2.2.0' or later but this fix will require some code change since the upgrade cause to build fail due to compilation errors like: 'src/main/java/com/datastax/mgmtapi/resources/LifecycleResources.java:[425,28] cannot access com.fasterxml.jackson.core.exc.StreamWriteException'"
```

Comment:
Just so I understand, https://github.com/advisories/GHSA-57j2-w4cx-62h2 is about `jackson-databind`... how does upgrading `swagger-jaxrs2` come into play?

Reply:
I was using the `mvn dependency:tree` command to figure the dependencies out , according to the following tree:


```shell
[INFO] +- io.swagger.core.v3:swagger-jaxrs2:jar:2.1.6:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.11.1:compile
[INFO] |  +- io.github.classgraph:classgraph:jar:4.8.65:compile
[INFO] |  +- org.javassist:javassist:jar:3.22.0-GA:compile
[INFO] |  +- io.swagger.core.v3:swagger-models:jar:2.1.6:compile
[INFO] |  +- io.swagger.core.v3:swagger-annotations:jar:2.1.6:compile
[INFO] |  +- io.swagger.core.v3:swagger-integration:jar:2.1.6:compile
[INFO] |  |  \- io.swagger.core.v3:swagger-core:jar:2.1.6:compile
[INFO] |  |     +- jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2:compile
[INFO] |  |     +- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:jar:2.11.1:compile
[INFO] |  |     |  \- org.yaml:snakeyaml:jar:1.26:compile
[INFO] |  |     \- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:jar:2.11.1:compile
[INFO] |  \- com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:jar:2.11.1:compile
[INFO] |     +- com.fasterxml.jackson.jaxrs:jackson-jaxrs-base:jar:2.11.1:compile
[INFO] |     \- com.fasterxml.jackson.module:jackson-module-jaxb-annotations:jar:2.11.1:compile
[INFO] |        \- jakarta.activation:jakarta.activation-api:jar:1.2.1:compile
```

I assumed I should upgrade `swagger-jaxrs2` to fix the CVEs related to the `jackson-databind`, does that makes sense?

Reply:
@stormqueen1990 maybe could help me on this if there is any misunderstanding from my side about how to fix CVEs in these types of maven deps?

Reply:
I concur with @developer-guy. There is no other dependency of this project on `jackson-databind` apart from this transitive one coming through `swagger-jaxrs2`.

Reply:
Okay, great. Thanks to both of you!